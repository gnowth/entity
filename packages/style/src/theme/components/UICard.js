import { css } from 'styled-components';

import { boxshadow, color, variable } from '../../selectors';

export const component_uiCard_danger = {
  css: css`
    color: ${color({ name: 'danger' })};
    background-color: ${color({ name: 'danger', weight: '50' })}
  `,
};

export const component_uiCard_page = {
  css: css`
    border: 0;
    margin: auto;
    max-width: ${variable({ name: 'size', variant: 'pageMaxWidth' })};
  `,
};

export const component_uiCard_page_flat = {
  css: css`
    border: 0;
    border-radius: 0;
  `,
};

export const component_uiCard_panel = {
  css: css`
    ${boxshadow({ name: 'material1' })}
    border: 0;
    border-bottom: 0.25rem solid ${color({ name: 'secondary' })};
  `,
};

export const component_uiCard_panel_focused = {
  css: css`
    ${boxshadow({ name: 'material1' })}
    background-color: ${color({ name: 'gray', weight: '100' })};
    border: 0;
    border-bottom: 0.25rem solid ${color({ name: 'secondary' })};
  `,
};
