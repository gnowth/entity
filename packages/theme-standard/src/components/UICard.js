import { boxshadow, color, sizeGridBase, variable } from '@gnowth/theme';
import { css } from 'styled-components';

export const component_uiCard_danger = {
  css: css`
    background-color: ${color({ palette: 'danger', paletteWeight: '50' })};
    border: 1px solid currentColor;
    border-radius: calc(${sizeGridBase} * 0.5);
    color: ${color({ palette: 'danger' })};
  `,
  padding: '1rem',
};

export const component_uiCard_page = {
  css: css`
    background-color: ${color({ palette: 'gray', paletteWeight: '400' })};
    margin: auto;
    max-width: ${variable({ name: 'size', variant: 'pageMaxWidth' })};
  `,
  padding: '2rem 0 4rem',
};

export const component_uiCard_page_flat = {
  padding: '2rem 0 4rem',
};

export const component_uiCard_panel = {
  css: css`
    ${boxshadow({ name: 'material1' })}
    background-color: white;
  `,
  padding: '2rem',
};

export const component_uiCard_panel_focused = {
  css: css`
    ${boxshadow({ name: 'material1' })}
    background-color: ${color({ palette: 'gray', paletteWeight: '100' })};
    border-bottom: 0.25rem solid ${color({ palette: 'secondary' })};
  `,
  padding: '2rem',
};

export const component_uiCard_screen = {
  padding: '1rem',
};
