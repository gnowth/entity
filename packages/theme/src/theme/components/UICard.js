import { css } from 'styled-components';

import { boxshadow, color, sizeGridBase, variable } from '../../selectors';

export const component_uiCard_danger = {
  css: css`
    background-color: ${color({ palette: 'danger', paletteWeight: '50' })};
    border: 1px solid currentColor;
    border-radius: calc(${sizeGridBase} * 0.5);
    color: ${color({ palette: 'danger' })};
  `,
  padding: '1rem',
};

export const component_uiCard_page = {
  css: css`
    background-color: ${color({ palette: 'gray', paletteWeight: '400' })};
    margin: auto;
    max-width: ${variable({ name: 'size', variant: 'pageMaxWidth' })};
  `,
  padding: '1rem',
};

export const component_uiCard_page_flat = {
  padding: '1.5rem',
};

export const component_uiCard_panel = {
  css: css`
    ${boxshadow({ name: 'material1' })}
    border-bottom: 0.25rem solid ${color({ palette: 'secondary' })};
  `,
  padding: '1rem',
};

export const component_uiCard_panel_focused = {
  css: css`
    ${boxshadow({ name: 'material1' })}
    background-color: ${color({ palette: 'gray', paletteWeight: '100' })};
    border-bottom: 0.25rem solid ${color({ palette: 'secondary' })};
  `,
  padding: '1rem',
};

export const component_uiCard_screen = {
  padding: '1rem',
};
