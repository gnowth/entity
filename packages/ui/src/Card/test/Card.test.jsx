import 'jest-styled-components';
import renderer from 'react-test-renderer';
import React from 'react';

import UICard from '../Card';

it('renders correctly', () => {
  const tree = renderer.create(
    <UICard />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
