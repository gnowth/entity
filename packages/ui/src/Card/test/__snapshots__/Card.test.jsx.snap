// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  background-color: black;
  color: black;
}

<div
  className="c0"
/>
`;
